implicits/src-0/upickle/implicits/macros.scala

```diff
@@ -8,35 +8,59 @@ def getDefaultParmasImpl[T](using qctx: QuoteContext,
   tpe: Type[T]): Expr[Map[String, AnyRef]] =
   import qctx.tasty._
   val sym = tpe.unseal.symbol
-  val comp = sym.companionClass
-  val names =
-    for p <- sym.caseFields if p.flags.is(Flags.HasDefault)
-    yield p.name
-  val namesExpr: Expr[List[String]] =
-    Expr.ofList(names.map(Expr(_)))
-
-  val body = comp.tree.asInstanceOf[ClassDef].body
-  val idents: List[Ref] =
-    for case deff @ DefDef(name, _, _, _, tree) <- body
-    if name.startsWith("$lessinit$greater$default")
-    yield Ref(deff.symbol)
-  val identsExpr: Expr[List[Any]] =
-    Expr.ofList(idents.map(_.seal.cast[Any]))
-
-  '{ $namesExpr.zip($identsExpr.map(_.asInstanceOf[AnyRef])).toMap }
+
+  if (sym.isClassDef) {
+    val comp = if (sym.isClassDef) sym.companionClass else sym
+    val names =
+      for p <- sym.caseFields if p.flags.is(Flags.HasDefault)
+      yield p.name
+    val namesExpr: Expr[List[String]] =
+      Expr.ofList(names.map(Expr(_)))
+
+    val body = comp.tree.asInstanceOf[ClassDef].body
+    val idents: List[Ref] =
+      for case deff @ DefDef(name, _, _, _, tree) <- body
+      if name.startsWith("$lessinit$greater$default")
+      yield Ref(deff.symbol)
+    val identsExpr: Expr[List[Any]] =
+      Expr.ofList(idents.map(_.seal.cast[Any]))
+
+    '{ $namesExpr.zip($identsExpr.map(_.asInstanceOf[AnyRef])).toMap }
+  } else {
+    '{ Map.empty }
+  }
 end getDefaultParmasImpl
 
-inline def summonAll[T <: Tuple]: List[_] =
+inline def summonList[T <: Tuple]: List[_] =
```

Comment:
summonAll was defined here temporarily while we didn't have one in Dotty. Now we do. Can we completely remove summonAll and summonList from here and use the Dotty implementation?